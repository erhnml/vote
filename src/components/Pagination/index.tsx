import React from 'react';
import styled from 'styled-components';

interface PaginationProps {
  changeStep: (step:number) => void,
  currentPage: number,
  total:number
}

interface StyledProps {
  active?: boolean
}

const Pagination = ({total, changeStep, currentPage}: PaginationProps) => {
  const totalPage = Math.ceil(total / 5);
  return (
    <Wrapper>
      {currentPage > 1 &&  (
        <Step onClick={() => changeStep(currentPage - 1)}>
          Prev
        </Step>
      )}
      {[...Array(totalPage)].map((e, i) => {
        const pageNumber = i + 1;
        return (
          <Step
            key={i}
            onClick={() => changeStep(pageNumber)} 
            active={pageNumber === currentPage}
          >
          <span>{pageNumber}</span>
        </Step>
        )
      })}
      {currentPage < totalPage && (
        <Step onClick={() => changeStep(currentPage + 1)}>
          Next
        </Step>
      )}
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 30px;
`;
export const Step = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  width: 40px;
  height: 40px;
  margin-right: 10px;
  cursor: pointer;
  ${({active}:StyledProps) => active && `
    background-color: black;
    color: white;
  `}
`;
export default Pagination;
