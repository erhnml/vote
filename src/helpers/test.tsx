import React from 'react';
import Theme from '../theme';
import { mount } from 'enzyme';

export function mountWithTheme(child: React.ReactElement) {
  return mount(child, {
      wrappingComponent: ({ children }) => <Theme>{children}</Theme>,
  });
}
