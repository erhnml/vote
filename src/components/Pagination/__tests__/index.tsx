
import React from 'react';

import Pagination, { Step } from '../index';
import { mountWithTheme } from '../../../helpers/test';

describe('Pagination', () => {
  it('renders correctly', () => {
    expect(
      <Pagination 
        changeStep={() =>{}}
        currentPage={1}
        total={10}
      />
    ).toMatchSnapshot();
  });

  it('mount correctly', () => {
    expect(() => mountWithTheme(
      <Pagination 
        changeStep={() =>{}}
        currentPage={1}
        total={10}
      />
    )).not.toThrow();
  });

  it('should trigger event correctly', () => {
    const mockFn = jest.fn();
    const wrapper = mountWithTheme(
      <Pagination 
        changeStep={mockFn}
        currentPage={2}
        total={20}
      />
    );
    const step = wrapper.find(Step);
    step.at(0).simulate('click')
    step.at(1).simulate('click')
    step.at(5).simulate('click')
    expect(mockFn.mock.calls.length).toBe(3);
  });

  it('should have correct step count', () => {
    const wrapper = mountWithTheme(
      <Pagination 
        changeStep={() => {}}
        currentPage={2}
        total={127}
      />
    );
    expect(wrapper.find(Step).length).toBe(28)
  })
  
})
