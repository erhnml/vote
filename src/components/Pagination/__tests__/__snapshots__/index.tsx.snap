// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<Pagination
  changeStep={[Function]}
  currentPage={1}
  total={10}
/>
`;
